src/io/dup.rs

```diff
@@ -78,3 +78,36 @@ pub fn dup2<Fd: AsFd>(fd: Fd, new: &mut OwnedFd) -> io::Result<()> {
 pub fn dup3<Fd: AsFd>(fd: Fd, new: &mut OwnedFd, flags: DupFlags) -> io::Result<()> {
     backend::io::syscalls::dup3(fd.as_fd(), new, flags)
 }
+
+/// Utility function to safely `dup2` over stdin (fd 0).
+#[cfg(not(any(windows, target_os = "wasi")))]
+#[inline]
+#[allow(unsafe_code)]
```

Comment:
Could we move these functions to src/io/stdio.rs? I think "stdio-using things" is the higher-level grouping here, than "dup-related things". And, that file already has a `#![allow(unsafe_code)]` so we wouldn't need these per-function attributes.